lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+            else:
+                pass
+
+        return create
+
+
+    def update_create_params(self, create):
+
+        if not create:
+            return create
+
+        v_template = 'Default_VRF_Universal' if not create.get('vrf_template') else create['vrf_template']
+        ve_template = 'Default_VRF_Extension_Universal' if not create.get('vrf_extension_template') \
+            else create['vrf_extension_template']
+        src = None if not create.get('source') else create['source']
+        s_v_template = None if not create.get('service_vrf_template') else create['service_vrf_template']
+
+        create_upd = dict()
+        create_upd.update({'fabric': self.fabric})
+        create_upd.update({'vrfName': create['vrf_name']})
+        create_upd.update({'vrfTemplate': v_template})
+        create_upd.update({'vrfExtensionTemplate': ve_template})
+        create_upd.update({'vrfId': create['vrf_id']})
+        create_upd.update({'serviceVrfTemplate': s_v_template})
+        create_upd.update({'source': src})
+
+        template_conf = dict()
+        template_conf.update({'vrfSegmentId': create['vrf_id']})
+        template_conf.update({'vrfName': create['vrf_name']})
+
+        create_upd.update({'vrfTemplateConfig': json.dumps(template_conf)})
 
-    return False
+        return create_upd
 
 
-def vrf_create_payload(module):
+    def get_have(self):
+        """
 
-    payload = dict()
-    payload['fabric'] = module.params['fabric']
-    payload['vrfName'] = module.params['vrf_name']
-    payload['vrfTemplate'] = module.params['vrf_template']
-    payload['vrfExtensionTemplate'] = module.params['vrf_extension_template']
-    payload['vrfTemplateConfig'] = module.params['vrf_template_config']
-    payload['vrfId'] = module.params['vrf_id']
+        Sufficient error checking??
+        Possiibe inputs from DCNM:
+        1. API failures
+        2. Blank responses - no vrfs
+                           - no attachments
+                           - no deployments
+        3. typecast booleans before checks?
+        """
 
-    json_data = json.dumps(payload)
+        have_create = list()
+        have_attach = list()
+        have_deploy = dict()
 
-    return json_data
+        all_vrfs = ''
 
+        method = 'GET'
+        path = '/rest/top-down/fabrics/{}/vrfs'.format(self.fabric)
+
+        resp = dcnm_send(self.module, method, path)
+
+        if resp.get('ERROR') == 'Not Found' and resp.get('RETURN_CODE') == 404:
+            logit("Fabric not found")
+            return have_create, have_attach, have_deploy
+
+        if not resp['DATA']:
+            return have_create, have_attach, have_deploy
+
+        for vrf in resp['DATA']:
+            all_vrfs += vrf['vrfName'] + ','
+
+        path = '/rest/top-down/fabrics/{}/vrfs/attachments?vrf-names={}'.format(self.fabric, all_vrfs[:-1])
+
+        all_vrf_attach = dcnm_send(self.module, method, path)
+
+        for vrf in resp['DATA']:
+            t_conf = dict()
```

Comment:
What does `t_conf` mean?

Reply:
template_config which needs to be part of vrf create payload, repetitive information, but mandated by payload.